### ISSUE TITLE
XDMCP support

### ISSUE THREAD

**Issue by bitstreamout** [sddm/sddm#1180]
<cite>
Important: Supported Display Managers
A machine can reliably accept VNC connections only if it uses a display manager that supports the XDMCP protocol. While gdm, lxdm, or lightdm support XDMCP, the KDE 5 default display manager sddm does not support it. When changing the default display manager, remember to log out of the current X session and restart the display manager with.
</cite>

it is extremly annoying that the default display manager is not able to handle the well known XDMCP protocol.  It should be noted that XDMCP is well understood and secure therefore there is no reason to ignore this feature.

**Comment by buzzra** [comment]
This is definitely required. I don't understand how a "new" wm was created without it. This shouldn't be a 'feature request' it is a bug/oversight that needs to be corrected before any serious use of sddm is considered.  KDE is useless without it.

**Comment by jonesmz** [comment]
Considering lots of people have been using sddm for "serious" things for many years, I think your statement may have a touch of hyperbole.

**Comment by buzzra** [comment]
Guess it depends on your definition of serious. I've been using XDMCP since before Linux was created to get work done from all kinds of thin and thick client devices. Guess I shouldn't be trying to use KDE anymore. Sad.

**Comment by jonesmz** [comment]
Don't get me wrong, I would love to see SDDM improved, and XDMCP sounds like an improvement.

Personally I do not use KDE, so I did not know that there was an issue.

What would it take to add XDMCP support?

**Comment by jonesmz** [comment]
Looking into it further, there seem to be pretty significant protocol issues with XDMCP, security and otherwise. It's not clear to me that anyone should be using it...?

**Comment by adojck** [comment]
In my opinion adding XDMCP support would be waste of time and effort.
* This protocol soon will be completely irrelevant
* Not a lot of people uses this functionality

**Comment by bitstreamout** [comment]
Hmmm ... what is the replacement of XDMCP?  And what is `not a lot of people'  ... clearly if no one shows that this protocol is useful and how/can be used even at home with serval systems (one big box and several small/thin clients) than clearly  ther would be more people using it.  Better then vnc

**Comment by adojck** [comment]
You can check this:
https://github.com/sddm/sddm/issues/1069
And the core maintainer of SDDM has similar opinion:
![image](https://user-images.githubusercontent.com/4030692/63593967-84d7d300-c5b5-11e9-9970-fab2efcea0e1.png)
This was on PR: https://github.com/sddm/sddm/pull/107

**Comment by davidedmundson** [comment]
> KDE is useless without it.

Note that KDE is not in any way dependent on SDDM.

If you require classic functionality, classic DMs exist and cover that usecase well.

**Comment by bitstreamout** [comment]
> You can check this:
> #1069
> And the core maintainer of SDDM has similar opinion:
> ![image](https://user-images.githubusercontent.com/4030692/63593967-84d7d300-c5b5-11e9-9970-fab2efcea0e1.png)
> This was on PR: #107

This is NOT the answer of my question:  what is the replacement of XDMCP? 
That upstream is not willingly to support a common feature of DM is clear to me ... nevertheless IMHO this is plain wrong and breaks a lot here (as sddm is default for KDE here).

**Comment by bitstreamout** [comment]
> > KDE is useless without it.
> 
> Note that KDE is not in any way dependent on SDDM.
> 
> If you require classic functionality, classic DMs exist and cover that usecase well.

xdm is somehow outof date and GNOME gdm is somehow not what I want. Again: what is wrong with XDMCP?  Why is upstream not willingly to add a major DM feature like XDMCP is?

**Comment by bitstreamout** [comment]
Very `nice' this discussion :(
NO explanatory statement, nothing only platitudes

**Comment by buzzra** [comment]
WOW GREAT support and discussion! We don't like it so we'll just close it. 
Still no answer to @bitstreamout question about what is the replacement for XDMCP in SDDM. 
More of a reason to start moving users away from KDE.

**Comment by davidedmundson** [comment]
> More of a reason to start moving users away from KDE.

These are not remotely related topics. bistreamout listed a bunch of DMs with XDMCP features in the beginning. KDE will work on any of them.

XDMCP is outdated and will not work on wayland a target direction for all the current desktops that use  SDDM.  I am happy to put time into guest sessions to launch thin VNC clients or anything else.
Neither pier or I will write new code for something already outdated when perfectly good alternatives for users who need XDMCP exist.

If there is a reason why those alternatives are not suitable and there is a problem to be solved, I will be here to help.

**Comment by bitstreamout** [comment]
If wayland is without XDMCP aka no secure remote network connections I guess(?) it is a clear step backward and can not be the future of the X Window system.  Simply becasue this would be  not usable in professional environments with thick server and thin X clients

**Comment by bitstreamout** [comment]
(Tiger)VNC with systemd on demand is doable for ONE user yes ... to scale this up for a thin client farm this becomes complicated ... beside this VNC (X) server^H^H^H^H^H^Hwindow is a poor replacement for a real driven X sever

**Comment by jonesmz** [comment]
@davidedmundson 

If you do end up adding support for things like launching a per-user remote session, I would ask you to add freerdp-shadow-cli to your initial discovery investigation.

This project is related: https://github.com/ogon-project/ogon-project

**Comment by blazingraptor** [comment]
I'll try and do it, first
* Gotta finish my Systems Admin classes first
* Gotta learn whatever language this is written in
* I may need some help, I struggle with references and pointers.
* Experience, .NET/C#, Javascript, SQL, Bash, CSS, PHP.
* I'm transitioning from Web/Windows development to Linux Systems Admin.

**Comment by jonesmz** [comment]
sddm is primarily written in Qt (C++), it also uses the QML markup language from the Qt project to drive the graphical user interface.

I'm sure that if you ask people questions here, someone will be able to provide guidance and feedback.

**Comment by Spanni26** [comment]
Here we are running about 300 diskless clients which are using different servers. And we are definitely not the only ones. So sddm seems to be only for home desktop users.

**Comment by agaida** [comment]
Hmm - i don't understand the problem to be honest - sddm is just a display manager - so if you are not satisfied with - why don't you use just another one? All the other arguments are strawmans too  imho.

**Comment by akorn** [comment]
> Hmm - i don't understand the problem to be honest - sddm is just a display manager - so if you are not satisfied with - why don't you use just another one? All the other arguments are strawmans too imho.

The problem, as I understand it, is that the other display managers that do support xdmcp are no longer being actively maintained (maybe with the exception of gdm?). Official word from desktop environment developers is to use sddm, which further contributes to interest in other DMs declining. Since the other DMs are becoming less and less relevant, DEs feel more and more justified in recommending sddm, and in the end it will become nearly impossible (as in very tedious) to use anything else.

Of course, if nobody with the ability to implement xdmcp support in sddm (or to maintain one of the other display managers) is actually willing to do it, xdmcp will just die, and people who depended on it will be left hanging.

Since xdmcp is mostly relevant for organizations, those organizations can take a look at how much xdmcp going away would cost them, and invest 5-10% of that sum into xdmcp related open source development. Not a terribly likely outcome, I admit.

I think it's entirely reasonable and fair that the people working on sddm for free only work on aspects that they personally find useful or interesting, and badgering them to work on xdmcp instead isn't going to help.

On the whole, this is all very unfortunate and sad, but it can't (feasibly) be helped.

**Comment by trantor** [comment]
Sad indeed. If my memory helps me correctly, X was developed in a networked world of mainframes and remote connections, which gave it features such as XDMCP. At the time of Wayland such things were deemed as bloated and mostly useless, which I guess is a reflection of the current, mostly single-user, user environment.
I feel that often the basic needs of the majority trump the more sophisticated needs of the few, resulting in less flexible products. Massification and all. Too often I've seen arguments such as "Let's get rid of that. Noone/nobody I know of uses it" (apart from those who do...) Sad it is, truly.

**Comment by cornfeedhobo** [comment]
@adojck It's been 3 years of SDDM claiming that XDMCP will be irrelevant soon. I guess I'll be back to comment in 2022 :(

**Comment by igorfk** [comment]
Hi,
I know, it is closed, no XDMCP for SDDM.
But does anyone have a suggestion on how could I do the following without XDMCP in SDDM?
In the school I work we configured the desktops to multiseat to two more seats on each computer (one computer, with one gpu and 3 outputs, for 3 students at the same time), local chrooted Xephyrs opens a connection to the local XDMCP, the Xephyrs are positioned fullscreen at separated screens with respective keyboard and mouse, a commom configuration if you search for "Xephyr multiseat" over the internet.
I had tried with local VNC (horrible speeds, even with sockets and tweaked a lot is not comparable to XDMCP)
Sorry if this is the wrong place to ask, as I stated before I know this functionality will not happen in SDDM, but I'm searching for alternatives a long time.
